general/development/policies.md

```diff
@@ -0,0 +1,211 @@
+
+:::
+
+## Moodle architecture
+
+Moodle tries to run on the widest possible range of platforms, for the widest possible number of people, while remaining easy to install, use, upgrade and integrate with other systems.
+
+:::info
+
+For more about this, see [Moodle architecture](https://docs.moodle.org/dev/Moodle_architecture).
+
+:::
+
+## Plugins
+
+Moodle has a general philosophy of modularity.  There are nearly 30 different standard types of plugins and even more sub-plugin types, however all of these plugin types work the same way. Blocks and activities are the only small exceptions.
+
+:::info
+
+For more about this, see [Moodle plugins](https://docs.moodle.org/dev/Plugins) and [Moodle sub-plugins](https://docs.moodle.org/dev/Subplugins).
+
+:::
+
+## Coding style
+
+Consistent [coding style](/general/development/policies/codingstyle) is important in any development project, and particularly so when many developers are involved. A standard style helps to ensure that the code is easier to read and understand, which helps overall quality.
+
+Writing your code in this way is an important step to having your code accepted by the Moodle community.
+
+:::info
+
+For more about this, see the [Moodle coding style](/general/development/policies/codingstyle).
+
+:::
+
+## Security
+
+Security is about protecting the interests and data of all our users.  Moodle may not be banking software, but it is still protecting a lot of sensitive and important data such as private discussions and grades from outside eyes (or student hackers!) as well as protecting our users from spammers and other internet predators.
+
+It's also a script running on people's servers, so Moodle needs to be a responsible Internet citizen and not introduce vulnerabilities that could allow crackers to gain unlawful access to the server it runs on.
+
+:::info
+
+Any single script (in Moodle core or a third party module) can introduce a vulnerability to thousands of sites, so it's important that all developers strictly follow our [Moodle security guidelines](/general/development/policies/security).
+
+:::
+
+## HTML5 and CSS
+
+It's important that Moodle produces strict, well-formed [HTML 5](http://en.wikipedia.org/wiki/HTML5) code (preferably backwards compatible with XHTML 1.1 if possible), compliant with all common accessibility guidelines (such as [W3C WAG 2.0](http://www.w3.org/TR/WCAG20/), [ARIA](http://www.w3.org/TR/wai-aria-practices/)).
+
+CSS should be used for layout. Moodle comes with several themes installed. Beginning with version 2.7, only the 'Clean' theme comes in the base Moodle code. The 'standard' theme, which should be a plain theme suitable to act as a building block for other themes. That should contain the minimal styling to make Moodle look OK and be functional. Then Moodle comes with several other default themes that look good and demonstrate various techniques for building themes.
```

Comment:
THis needs updating for Boost/Classic vs. Clean + Standard.

```suggestion
CSS should be used for layout. Moodle comes with several themes installed. Beginning with version 2.7, only the 'Boost, and 'Classic' themes are included with the base Moodle code.

:::note Writing your own theme

We recommend that if you are writing your own theme that it should extend the Moodle 'Boost' theme.

:::
```

The rest seems to be irrelevant now I think.